app/configurator/interactive-filters/time-slider.tsx

```diff
@@ -12,57 +12,29 @@ import { useTimeFormatUnit } from "@/formatters";
 import { DimensionMetadataFragment, TimeUnit } from "@/graphql/query-hooks";
 import { TemporalDimension } from "@/graphql/resolver-types";
 import { Icon } from "@/icons";
+import { Timeline } from "@/utils/observables";
 import useEvent from "@/utils/use-event";
 
-// TODO: make this configurable
-const ANIMATION_DURATION = 10000;
-
 type SliderDatum = {
   ms: number;
   formattedDate: string;
 };
 
-type TimeSliderState = {
-  progress: number;
-  setProgress: React.Dispatch<React.SetStateAction<number>>;
-  animating: boolean;
-  setAnimating: React.Dispatch<React.SetStateAction<boolean>>;
-};
+const TimelineContext = React.createContext<Timeline | undefined>(undefined);
 
-const TimeSliderStateContext = React.createContext<TimeSliderState>({
-  progress: 0,
-  setProgress: () => {},
-  animating: false,
-  setAnimating: () => {},
-});
+const useTimeline = () => {
+  // This is a "hack" to force re-rendering of the component using the hook.
+  const [_, setS] = React.useState(0);
+  const timeline = React.useContext(TimelineContext);
 
-const useTimeSliderState = () => {
-  const ctx = React.useContext(TimeSliderStateContext);
+  React.useEffect(() => {
+    const callback = () => setS((s) => s + 1);
+    timeline?.subscribe(callback);
 
-  if (!ctx) {
-    throw new Error(
-      "You need to wrap your component in <TimeSliderStateProvider /> to useTimeSliderContext()"
-    );
-  }
+    return () => timeline?.unsubscribe(callback);
+  }, [timeline]);
```

Comment:
I think it could be a bit improved by not tying it to the timeline.

Maybe have a useObservable(observable: Observable) and then a useTimeline that uses this and tying it to the context.